bodo/pandas/plan_optimizer.pyx

```diff
@@ -28,11 +228,22 @@ cdef extern from "duckdb/common/enums/join_type.hpp" namespace "duckdb" nogil:
         RIGHT_ANTI "duckdb::JoinType::RIGHT_ANTI"
 
 
-cdef extern from "duckdb/planner/operator/logical_comparison_join.hpp" namespace "duckdb" nogil:
-    cdef cppclass CLogicalComparisonJoin" duckdb::LogicalComparisonJoin"(CLogicalOperator):
-        CLogicalComparisonJoin(CJoinType join_type)
+cdef extern from "duckdb/planner/operator/logical_join.hpp" namespace "duckdb" nogil:
```

Comment:
Why is `LogicalJoin` necessary to be wrapped? The Cython code doesn't need to match the C++ side exactly. I think it's best to only wrap what we need.

Reply:
Not a cython expert and the info I was seeing said they had to match.  I tried it your way and it seemed to work.  Personally, I don't mind adding stuff that is temporarily (or in hindsight even permanently) not used because I consider it future proofing when something in the future may require it.  I think we have different philosophies here.

Reply:
Matching just the class hierarchy is ok and maybe even better for when other engineers read the code. But other unused code can be confusing for engineers new to the project.